java/ql/src/Telemetry/DatabaseQualityDiagnostics.ql

```diff
@@ -0,0 +1,40 @@
+/**
+ * @name Low Java analysis quality
+ * @description Low Java analysis quality
+ * @kind diagnostic
+ * @id java/diagnostic/database-quality
+ */
+
+import java
+import DatabaseQuality
+
+private newtype TDbQualityDiagnostic =
+  TTheDbQualityDiagnostic() {
+    exists(float percentageGood |
+      CallTargetStatsReport::percentageOfOk(_, percentageGood)
+      or
+      ExprTypeStatsReport::percentageOfOk(_, percentageGood)
+    |
+      percentageGood < 95
+    )
+  }
+
+class DbQualityDiagnostic extends TDbQualityDiagnostic {
+  string toString() {
+    result =
+      "There were significant issues scanning Java code. " +
+        "This may be caused by problems identifying dependencies or use of generated source code, among other reasons -- "
+        + "see other CodeQL diagnostics reported here for more details of possible causes. " +
+        "This may lead to false-positive or missing results. Consider analysing Java using the `autobuild` or `manual` build modes."
```

Comment:
```suggestion
        "This may lead to false-positive or missing results. Consider addressing these warnings or scanning Java using the `autobuild` or `manual` build modes."
```

Indicate that enhancements can be made with `build-mode: none`  (ex: if you can register package auth credentials or switch to self-hosted with registry access).   Presuming we are likely to have mostly similar issues in autobuild/manual mode but with a harder failure.  Avoid `analyse` vs `analyze`.

Reply:
And if you want, as @felickz suggested in another comment, you could definitely link to the [build modes documentation](https://docs.github.com/en/code-security/code-scanning/creating-an-advanced-setup-for-code-scanning/codeql-code-scanning-for-compiled-languages#codeql-build-modes). I suppose you could either directly link on the word `build modes` or just add a sentence at the end saying something like "For more information, see [the documentation on CodeQL build modes](https://docs.github.com/en/code-security/code-scanning/creating-an-advanced-setup-for-code-scanning/codeql-code-scanning-for-compiled-languages#codeql-build-modes)."